src/main/scala/tech/cryptonomic/conseil/tezos/ApiOperations.scala

```diff
@@ -124,17 +150,48 @@ object ApiOperations {
     * @return       List of blocks
     */
   def fetchBlocks(filter: Filter): Try[Seq[Tables.BlocksRow]] = Try {
-    val action = for {
-      b: Tables.Blocks <- Tables.Blocks
-      og <- Tables.OperationGroups
-      if b.hash === og.blockId &&
-      filterBlockIDs(filter, b) &&
-      filterBlockLevels(filter, b) &&
-      filterChainIDs(filter, b) &&
-      filterProtocols(filter, b) &&
-      filterOperationIDs(filter, og) &&
-      filterOperationSources(filter, og)
-    } yield (b.chainId, b.protocol, b.level, b.proto, b.predecessor, b.validationPass, b.operationsHash, b.protocolData, b.hash, b.timestamp, b.fitness)
+
+    val filteredBlocks = Tables.Blocks.filter({block =>
+        filterBlockIDs(filter, block) &&
+        filterBlockLevels(filter, block) &&
+        filterChainIDs(filter, block) &&
+        filterProtocols(filter, block)})
+    val filteredOpGroups = Tables.OperationGroups.filter({opGroup =>
+        filterOperationIDs(filter, opGroup) &&
+        filterOperationSources(filter, opGroup) &&
+        filterOperationGroupKinds(filter, opGroup)})
+    val filteredOps = Tables.Operations.filter({op => filterOperationKinds(filter, op)})
+
+    val isOGFilter = isOperationGroupFilter(filter)
+    val isOFilter = isOperationFilter(filter)
```

Comment:
Furthermore, since these two variables are being used only once, can we directly use the functions in the tuple on which we pattern match?